tests/providers/amazon/aws/hooks/test_base_aws.py

```diff
@@ -112,10 +112,20 @@ def test_get_client_type_returns_a_boto3_client_of_the_requested_type(self):
         if client.list_clusters()['Clusters']:
             raise ValueError('AWS not properly mocked')
 
-        hook = AwsBaseHook(aws_conn_id='aws_default', client_type='emr')
-        client_from_hook = hook.get_client_type('emr')
-
-        assert client_from_hook.list_clusters()['Clusters'] == []
+        with self.subTest("client_type set in both class and function"):
+            hook = AwsBaseHook(aws_conn_id='aws_default', client_type='emr')
+            client_from_hook = hook.get_client_type(client_type='emr')
+            assert client_from_hook.list_clusters()['Clusters'] == []
+
+        with self.subTest("client_type set only in class"):
+            hook = AwsBaseHook(aws_conn_id='aws_default', client_type='emr')
+            client_from_hook = hook.get_client_type()
+            assert client_from_hook.list_clusters()['Clusters'] == []
+
+        with self.subTest("client_type overwrite by function"):
+            hook = AwsBaseHook(aws_conn_id='aws_default', client_type='dynamodb')
+            client_from_hook = hook.get_client_type(client_type='emr')
+            assert client_from_hook.list_clusters()['Clusters'] == []
```

Comment:
Any reason these need to be subtests instead of standalone test functions?

Reply:
no special reason. I'll change it

Reply:
It’s OK either way, I was just curious. But it’s best to change it since subtests could introduce issues if this test class ever needs to be rewritten with to pytest-style.